multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+        value: ethereum_types::U256::zero(),
+        data: Bytes(data.to_vec()),
+        chain_id: Some(chain.rec().num_chain_id),
+    };
+    let signed_tx = primitive_sign_transaction(web3, tx, from_wallet);
+    match web3
+        .eth()
+        .send_raw_transaction(signed_tx.raw_transaction)
+        .wait()
+    {
+        Ok(tx_hash) => eprintln!(
+            "Transaction {:?} of {} wei of MASQ was sent from wallet {} to {}",
+            tx_hash, amount_minor, from_wallet, to_wallet
+        ),
+        Err(e) => panic!("Transaction for token transfer failed {:?}", e),
+    }
+}
+
+fn primitive_sign_transaction(
+    web3: &Web3<Http>,
+    tx: TransactionParameters,
+    signing_wallet: &Wallet,
+) -> SignedTransaction {
+    let secret = &signing_wallet
+        .prepare_secp256k1_secret()
+        .expect("wallet without secret");
+    web3.accounts()
+        .sign_transaction(tx, secret)
+        .wait()
+        .expect("transaction preparation failed")
+}
+
+fn transfer_transaction_fee_amount_to_address(
+    from_wallet: &Wallet,
+    to_wallet: &Wallet,
+    amount_minor: u128,
+    transaction_nonce: u64,
+    web3: &Web3<Http>,
+) {
+    let tx = TransactionRequest {
+        from: from_wallet.address(),
+        to: Some(to_wallet.address()),
+        gas: Some(*GAS_LIMIT),
+        gas_price: Some(*GAS_PRICE),
+        value: Some(ethereum_types::U256::from(amount_minor)),
+        data: None,
+        nonce: Some(ethereum_types::U256::try_from(transaction_nonce).expect("Internal error")),
+        condition: None,
+    };
+    match web3
+        .personal()
+        .unlock_account(from_wallet.address(), "", None)
+        .wait()
+    {
+        Ok(was_successful) => {
+            if was_successful {
+                eprintln!("Account {} unlocked for a single transaction", from_wallet)
+            } else {
+                panic!(
+                    "Couldn't unlock account {} for the purpose of signing the next transaction",
+                    from_wallet
+                )
+            }
+        }
+        Err(e) => panic!(
+            "Attempt to unlock account {:?} failed at {:?}",
+            from_wallet.address(),
+            e
+        ),
+    }
+    match web3.eth().send_transaction(tx).wait() {
+        Ok(tx_hash) => eprintln!(
+            "Transaction {:?} of {} wei of ETH was sent from wallet {:?} to {:?}",
+            tx_hash, amount_minor, from_wallet, to_wallet
+        ),
+        Err(e) => panic!("Transaction for token transfer failed {:?}", e),
+    }
+}
+
+fn assert_balances(
+    wallet: &Wallet,
+    blockchain_interface: &dyn BlockchainInterface,
+    expected_eth_balance: u128,
+    expected_token_balance: u128,
+) {
+    single_balance_assertion(
+        blockchain_interface,
+        wallet,
+        expected_eth_balance,
+        "ETH balance",
+        |blockchain_interface, wallet| blockchain_interface.get_transaction_fee_balance(wallet),
+    );
```

Comment:
You can reduce an argument (the last) in this function if you pass an enum instead of string `"ETH balance"`, and then later you can call the necessary function, which in this case is `get_transaction_fee_balance()`.


Another idea that you proposed was to migrate it to the new class that you intend you create `ExtendedBlockchainInterface`. I leave that decision in your hand.